src/cordova/build.action.mjs

```diff
@@ -33,35 +35,46 @@ export async function main(...parameters) {
   await runAction('cordova/setup', ...parameters);
 
   if (buildMode === 'debug') {
-    console.warn(`WARNING: building "${cordovaPlatform}" in [DEBUG] mode. Do not publish this build!!`);
+    console.warn(`WARNING: building "${outlinePlatform}" in [DEBUG] mode. Do not publish this build!!`);
   }
 
-  if (outlinePlatform === 'macos' || outlinePlatform === 'ios') {
-    const WORKSPACE_PATH = `${process.env.ROOT_DIR}/src/cordova/apple/${outlinePlatform}.xcworkspace`;
+  if (cordovaPlatform === 'osx' || cordovaPlatform === 'ios') {
+    const xcodebuildFlags = {
+      workspace: path.join(getRootDir(), 'src', 'cordova', 'apple', `${outlinePlatform}.xcworkspace`),
+      scheme: 'Outline',
+    };
+
     // TODO(fortuna): Specify the -destination parameter for build. Do we need it for archive?
+    await runXcodebuild.clean(xcodebuildFlags);
+
     if (buildMode === 'release') {
-      execSync(`xcodebuild -workspace ${WORKSPACE_PATH} -scheme Outline -configuration Release clean archive`, {
-        stdio: 'inherit',
+      return runXcodebuild.archive({
+        ...xcodebuildFlags,
+        configuration: 'Release',
       });
-    } else {
-      execSync(
-        `xcodebuild -workspace ${WORKSPACE_PATH} -scheme Outline -configuration Debug build CODE_SIGN_IDENTITY="" CODE_SIGNING_ALLOWED=NO`,
-        {
-          stdio: 'inherit',
-        }
-      );
     }
-    return;
+
+    process.env.CODE_SIGN_IDENTITY = '';
```

Comment:
These are not env variables. They are parameters of `xcodebuild build`. I think you'll need a way to call with arbitrary parameters. I'd let go the flags optimization, just be explicit about them: `runxcodeBuild.build('-workspace', WORKSPACE_PATH, ...)`

Reply:
This is probably what's breaking your build, btw